cinema/views.py

```diff
@@ -1 +1,64 @@
-# write views here
+from rest_framework import viewsets
+from django.db.models import QuerySet
+
+from cinema.models import Movie, MovieSession, CinemaHall, Actor, Genre
+from cinema.serializers import (
+    MovieSerializer,
+    MovieListSerializer,
+    MovieRetireveSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieSessionRetireveSerializer,
+    CinemaHallSerializer,
+    ActorSerializer,
+    GenreSerializer,
+)
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+
+    def get_serializer_class(self) -> MovieSerializer:
+        if self.action == "list":
+            return MovieListSerializer
+        if self.action == "retrieve":
+            return MovieRetireveSerializer
+        return MovieSerializer
+
+    def get_queryset(self) -> QuerySet:
+        queryset = self.queryset
+        if self.action in ("list", "retrieve"):
+            return queryset.prefetch_related("actors", "genres")
```

Comment:
You can provide multiple arguments into prefetch_related method. Instead of chaining `.prefetch_related('actors').prefetch_related('genres')`, you can write `.prefetch_related('actors', 'genres')`.